lib/azure_blob/client.rb

```diff
@@ -407,6 +407,10 @@ def put_blob_single(key, content, options = {})
       Http.new(uri, headers, signer:, **options.slice(:metadata, :tags)).put(content.read)
     end
 
+    def content_size(content)
+      content.bytesize
+    end
```

Comment:
This breaks the tests using StringIO. Here's what microsoft used in the old gem: https://github.com/Azure/azure-storage-ruby/blob/master/common/lib/azure/core/auth/shared_key.rb#L62-L66


```suggestion
      if content.respond_to?(:bytesize)
        content.bytesize
      else
        content.size
      end
```